drivers/adc/adc_mcux_adc16.c

```diff
@@ -252,19 +355,63 @@ static int mcux_adc16_init(const struct device *dev)
 #endif
 
 	ADC16_Init(base, &adc_config);
-#if defined(FSL_FEATURE_ADC16_HAS_CALIBRATION) && \
-	    FSL_FEATURE_ADC16_HAS_CALIBRATION
+#if defined(FSL_FEATURE_ADC16_HAS_CALIBRATION) &&                              \
+	FSL_FEATURE_ADC16_HAS_CALIBRATION
 	ADC16_SetHardwareAverage(base, kADC16_HardwareAverageCount32);
 	ADC16_DoAutoCalibration(base);
 #endif
 	if (config->channel_mux_b) {
 		ADC16_SetChannelMuxMode(base, kADC16_ChannelMuxB);
 	}
-	ADC16_EnableHardwareTrigger(base, false);
 
-	config->irq_config_func(dev);
+	if (IS_ENABLED(CONFIG_ADC_MCUX_ADC16_HW_TRIGGER)) {
+		ADC16_EnableHardwareTrigger(base, true);
+	} else {
+		ADC16_EnableHardwareTrigger(base, false);
+	}
+
 	data->dev = dev;
 
+	/* dma related init */
+#ifdef CONFIG_ADC_MCUX_ADC16_ENABLE_EDMA
+	/* Enable DMA. */
+	ADC16_EnableDMA(base, true);
+
+	data->adc_dma_config.dma_name = CONFIG_DMA_0_NAME;
```

Comment:
This should come from a devicetree API.  `DT_DMAS_LABEL*`

Reply:
thanks, updated to use DTS.